src/diffusers/models/unet_2d_blocks.py

```diff
@@ -317,6 +345,35 @@ def get_up_block(
             attn_num_head_channels=attn_num_head_channels,
             resnet_time_scale_shift=resnet_time_scale_shift,
         )
+    elif up_block_type == "KUpBlock2D":
+        return KUpBlock2D(
+            num_layers=num_layers,
+            in_channels=in_channels,
+            out_channels=out_channels,
+            temb_channels=temb_channels,
+            add_upsample=add_upsample,
+            resnet_eps=resnet_eps,
+            resnet_act_fn=resnet_act_fn,
+        )
+    elif up_block_type == "KCrossAttnUpBlock2D":
+        if cross_attention_dim is None:
+            raise ValueError("cross_attention_dim must be specified for KCrossAttnUpBlock2D")
+        is_first_block = in_channels == out_channels == temb_channels
+        return KCrossAttnUpBlock2D(
+            num_layers=num_layers,
+            in_channels=in_channels,
+            out_channels=out_channels,
+            temb_channels=temb_channels,
+            add_upsample=add_upsample,
+            resnet_eps=resnet_eps,
+            resnet_act_fn=resnet_act_fn,
+            cross_attention_dim=cross_attention_dim,
+            attn_num_head_channels=attn_num_head_channels,
+            attn1_type="self" if is_first_block else "cross",
+            attn2_type="cross" if is_first_block else None,
```

Comment:
Can we try to clean this up a bit more? I'm not too happy about the `attn2_type` variable in general. I think ideally it shouldn't exist at all

Reply:
✅